src/pages/billing/List/BillingProjectActions.ts

```diff
@@ -17,38 +19,39 @@ interface BillingProjectActionsProps {
 export const BillingProjectActions = (props: BillingProjectActionsProps) => {
   const [showDeleteProjectModal, setShowDeleteProjectModal] = useState(false)
   const [deleting, setDeleting] = useState(false)
+  const { workspaces, loading } = useWorkspaces()
 
-  return h(Fragment, [
-    h(MenuTrigger, {
-      closeOnClick: true,
-      side: 'right',
-      style: { marginRight: '0.5rem' },
-      content: h(Fragment, [
-        h(MenuButton, {
-          onClick: () => setShowDeleteProjectModal(true)
-        }, ['Delete Billing Project'])
-      ])
-    }, [
-      h(Link, { 'aria-label': 'Billing project menu', style: { display: 'flex', alignItems: 'center' } }, [
-        // @ts-ignore
-        icon('cardMenuIcon', { size: 16, 'aria-haspopup': 'menu' })
-      ])
-    ]),
-    showDeleteProjectModal && h(DeleteBillingProjectModal, {
-      projectName: props.projectName, deleting,
-      onDismiss: () => setShowDeleteProjectModal(false),
-      onConfirm: async () => {
-        setDeleting(true)
-        try {
-          await Ajax().Billing.deleteProject(props.projectName)
-          setShowDeleteProjectModal(false)
-          props.loadProjects()
-        } catch (err) {
-          reportError('Error deleting billing project.', err)
-          setShowDeleteProjectModal(false)
-        }
-        setDeleting(false)
+  const hasWorkspaces = _.find(
+    { namespace: props.projectName },
+    _.map('workspace', workspaces)
+  ) !== undefined
+
+  return div({ style: { marginLeft: 'auto' } }, [h(Link, {
+    disabled: loading || hasWorkspaces,
+    tooltip: Utils.cond(
+      [loading, () => 'Cannot delete billing project while workspaces are loading'],
+      [hasWorkspaces, () => 'Cannot delete billing project because it contains workspaces'],
+      () => `Delete billing project ${props.projectName}`
+    ),
+    style: { padding: '0.5rem' },
+    onClick: () => setShowDeleteProjectModal(true),
+  }, [icon('trash')]),
+
+  showDeleteProjectModal && h(DeleteBillingProjectModal, {
+    projectName: props.projectName, deleting,
+    onDismiss: () => setShowDeleteProjectModal(false),
+    onConfirm: async () => {
+      setDeleting(true)
```

Comment:
I wasn't able to unit test clicking the delete button because the anchor tag changes the window.location variable, which fails with the test DOM. I tried various things as suggested in https://github.com/jsdom/jsdom/issues/2112, but could get none of them to work.

We do have integration test coverage of actually clicking delete.